### ISSUE TITLE
Feature Request: Hot Corner for workspace overview

### ISSUE THREAD

**Issue by snaggen** [pop-os/cosmic-epoch#667]
Coming from Gnome Shell, I find myself throwing the mouse pointer to the top left corner all the time to open the workspace overview... I'm sure this might be a very common behaviour for us former Gnomers, so it would be a nice addition to have that in the Cosmic Desktop also.

**Comment by slackerbob** [comment]
I keep doing the same thing, and would love to have configurable hot corners in cosmic!

**Comment by nadir-ishiguro** [comment]
Should this feature be implemented, I think adding the option to make the corner or edge configurable would be very welcome, too. 

On Gnome, I've been using the extension called "Hot Edge" for that.

**Comment by centaurusgod** [comment]
This feature should be implemented. 
@mmstick

**Comment by GuyLiner** [comment]
Please :pray:

**Comment by Drakulix** [comment]
This is a good example of something we don't support right now (and that frankly isn't a huge priority atm), but that can easily provided by some external tool, given COSMIC just uses standard protocols.

Both [wlhc](https://git.sr.ht/~whynothugo/wlhc) and [waycorner](https://github.com/AndreasBackx/waycorner) should work just fine and provide this functionality.

**Comment by Shoryureppa01** [comment]
Please.  I use this as a reflex and my daily-driver thinkpad where I have the alpha release installed is getting tired of my grumbles when it doesn't work.

**Comment by eode** [comment]
I'm glad that this can be replicated with external programs, but sure would be nice to have native support.

**Comment by Xarius86** [comment]
I agree. I'm glad external tools can handle it, but having native options would be nice.

**Comment by Paulemeister** [comment]
For anyone too lazy to figure this out: I put together a systemd user service (place it in ~/.config/systemd/user/NAME.service) 
```
[Install]
WantedBy=cosmic-session.target

[Service]
ExecCondition=pgrep -xf cosmic-workspaces
ExecStart=waycorner
Restart=on-failure
RestartSec=2

[Unit]
After=cosmic-session.target
BindsTo=cosmic-session.target
Description=Service for  hotcorner on cosmic-epoch
```

with a waycorner config like this:  under ~/.config/waycorner/config.toml
```toml
[main-monitor]
enter_command = [  "cosmic-workspaces" ]
locations = [ "top_left"]  
size = 10  # default
margin = 20  # default
timeout_ms = 0  # default
color = "#FFFF0000"  # default
```
this should open and close the workspaces on touching the top left corner.

ISSUES: I had to have a workaround for the service, checking whether cosmic-workspaces was already running, otherwise it seems like pressing the workspaces button will start new instances of cosmic-workspaces that will not terminate. Feel free to improve upon this

EDIT: Edited WantedBy and After to start only on cosmic and only after cosmic has started.

**Comment by lethevimlet** [comment]
You can add hot corner and 3 finger swipe with this add-ons [https://github.com/lethevimlet/cosmic-applets](https://github.com/lethevimlet/cosmic-applets)

**Comment by Gibbz** [comment]
yes this would be great to see it implemented native. I use it on kde and gnome.
